### PR TITLE
Fiber/error handling

### PR DESCRIPTION
**Pull request by soupette**
- Implement `ErrorBoundary` component
- Fix several component css
- Fix `content-manager` model transition
- Fix `content-type-builder`plugin model creation bug in `content-manager` plugin
- Use `PopUpWarning` component in all plugins
- Created `Ico` and `IcoContainer` container components and implement them in `content-manager`and `content-type-builder` plugins
- Update documentation according to these changes

### CODE REVIEW

File: docs/3.x.x/en/plugins/ui-components.md [hunk 1]
```diff
@@ -66,6 +74,101 @@ export default Button;
 
 ***
 
+## Ico
+
+Ico components that works with fontAwesone.
```
Comment (Aurelsicoko) [inline docs/3.x.x/en/plugins/ui-components.md]: `fontAwesome`

File: docs/3.x.x/en/plugins/ui-components.md [hunk 2]
```diff
@@ -66,6 +74,101 @@ export default Button;
...
+
+### Usage
+
+| Property | Type | Required | Description |
+| -------- | ---- | -------- | ----------- |
+| `icoType` | string | no (default: `trash`) | fontAwesone ico name. Ex: <Ico icoType="pencil" /> |
```
Comment (Aurelsicoko) [inline docs/3.x.x/en/plugins/ui-components.md]: Typo `fontAwesome`

File: docs/3.x.x/en/plugins/ui-components.md [hunk 3]
```diff
@@ -66,6 +74,101 @@ export default Button;
...
+}
+
+export default FooPage;
+```
+
+***
+
+## IcoContainer
+
+Container containing two icons, generally used for editing or deleting data.
```
Comment (Aurelsicoko) [inline docs/3.x.x/en/plugins/ui-components.md]: Why `two`? `Many` is more appropriate

File: docs/3.x.x/en/plugins/ui-components.md [hunk 4]
```diff
@@ -214,17 +373,37 @@ class FooPage extends React.Component {
   }
 
   render() {
-    return(
+    const popupContent = {
+      cancel: "my-plugin.popup.danger.button.cancel",
```
Comment (Aurelsicoko) [inline docs/3.x.x/en/plugins/ui-components.md]: Single quotes would be appreciated.

> We should really find a way to apply a linter on the documentation examples.

File: packages/strapi-plugin-content-manager/admin/src/containers/App/reducer.js [hunk 5]
```diff
@@ -17,6 +17,12 @@ const initialState = fromJS({
 
 function appReducer(state = initialState, action) {
   switch (action.type) {
+    case EMPTY_STORE:
+      return state
```
Comment (Aurelsicoko) [inline packages/strapi-plugin-content-manager/admin/src/containers/App/reducer.js]:
```js
case EMPTY_STORE:
  return undefined;
```

I'm not sure if Immutable will allow this but based on [Dan Abramov's answer](https://stackoverflow.com/questions/35622588/how-to-reset-the-state-of-a-redux-store/35641992#35641992) it seems to be the right way.

File: packages/strapi-plugin-content-type-builder/admin/src/components/TableListRow/index.js [hunk 6]
```diff
@@ -38,19 +37,16 @@ class TableListRow extends React.Component { // eslint-disable-line react/prefer
...
-    e.stopPropagation();
-    this.setState({ showWarning: !this.state.showWarning });
-  }
+  handleShowModalWarning = () => this.setState({ showWarning: !this.state.showWarning });
 
   render() {
     const temporary = this.props.rowItem.isTemporary ? <FormattedMessage id="content-type-builder.contentType.temporaryDisplay" /> : '';
     const description = isEmpty(this.props.rowItem.description) ? '-' :  this.props.rowItem.description;
     const spanStyle = this.props.rowItem.isTemporary ? '60%' : '100%';
+    const icons = [{ icoType: 'pencil', onClick: this.handleEdit }, { onClick: this.handleShowModalWarning }];
```
Comment (Aurelsicoko) [inline packages/strapi-plugin-content-type-builder/admin/src/components/TableListRow/index.js]: I would prefer to say explicitly the second icon is a `trash`. There are many reasons:

- What happens if we decided to change the default icon in the future? You will have to come back here and update this line of code.
- It allows any developer to understand without reading precisely the documentation.

File: packages/strapi-plugin-content-manager/admin/src/containers/App/reducer.js (hunk 5 above)
Comment (soupette) [reply packages/strapi-plugin-content-manager/admin/src/containers/App/reducer.js]: I'll go with :
```js
case EMPTY_STORE:
  return initialState;
```

**Comment by Aurelsicoko** [comment]
LGTM 👍

**Pull request merged**